models/iris/train.py

```diff
@@ -0,0 +1,49 @@
+import time
+import random
+import pickle
+import warnings
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from certifai.common.utils.encoding import CatEncoder
+# supress all warnings
+warnings.filterwarnings('ignore')
+
+
+def main():
+    random.seed(0)
+    np.random.seed(0)
+
+    from sklearn import datasets
+
+    # Bring in test and training data.
+    x, y = datasets.load_iris(return_X_y=True)
+    x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=.5)
+
+    # Create an encoder
+    cat_columns = [] # All columns are numeric
+    encoder = CatEncoder(cat_columns, pd.DataFrame(x), normalize=True)
```

Comment:
I think only the training dataset (`x_train`) is supposed to be passed to the encoder here, to avoid leaking any info about the test dataset (and invalidate the model accuracy below).

Reply:
I double-checked with my resident expert, and its correct to pass the while dataset so that the encoder knows about all of the categories etc.